src/producer.cpp

```diff
@@ -143,7 +143,11 @@ void Producer::do_produce(const Message& message,
     const Buffer& payload = message.get_payload();
     const Buffer& key = message.get_key();
     const int policy = static_cast<int>(message_payload_policy_);
+#ifdef _USE_CPP17
+    int64_t duration = message.get_timestamp() ? message.get_timestamp().value().get_timestamp().count() : 0;
```

Comment:
I presume both optionals have `operator*` so rather than doing this you can maybe use that which should make the same expression work for both types?